client/store/cart.spec.js

```diff
@@ -0,0 +1,94 @@
+/* global describe beforeEach afterEach it */
+
+import {expect} from 'chai';
+import {findOrCreateCart, emptyCart, addingToCart, subtractFromCart} from './product';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import configureMockStore from 'redux-mock-store';
+import thunkMiddleware from 'redux-thunk';
+
+const middlewares = [thunkMiddleware];
+const mockStore = configureMockStore(middlewares);
+
+describe('RX Store - product - thunk creators', () => {
+	let store;
+	let mockAxios;
+    let fakeCartItems;
+    let userId;
+    let cartItem1;
+    let cartItem2;
+
+	const initialState = {
+		cartItems: []
+	};
+
+	beforeEach(() => {
+		mockAxios = new MockAdapter(axios);
+		store = mockStore(initialState);
+
+		cartItem1 = {
+			id: 1,
+			imageUrl: 'http://www.test.com/url.jpg',
+			name: 'Test Item1',
+			price: 23.99
+		};
+		cartItem2 = {
+			id: 2,
+			imageUrl: 'http://www.test.com/url.jpg',
+			name: 'Test Item2',
+			price: 99.99
+		};
+        fakeCartItems = [cartItem1, cartItem2];
+        userId = 2
+	});
+
+	afterEach(() => {
+		mockAxios.restore();
+		store.clearActions();
+	});
+
+	describe('findOrCreateCart', () => {
+		it('eventually dispatches the SET_CART action', async () => {
+            //TODO: mock axios call nees to be aligned with the real final route, (delete userId var if not needed)
+			mockAxios.onGet(`/api/carts/${userId}/active`).replyOnce(200, fakeCartItems);
```

Comment:
This route seems outdated. Does this mean this get request mock doesn't need to exist?

Reply:
Same with line 64